agent/charms/testflinger-agent-host-charm/tests/integration/test_charm_integration.py

```diff
@@ -0,0 +1,29 @@
+from pathlib import Path
+import pytest
+from pytest_operator.plugin import OpsTest
+
+# Root of the charm we need to build is two dirs up
+CHARM_PATH = Path(__file__).parent.parent.parent
+APP_NAME = "testflinger-agent-host"
+
+
+#@pytest.mark.skip_if_deployed
+#@pytest.mark.setup
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest):
+    charm = await ops_test.build_charm(CHARM_PATH)
+    app = await ops_test.model.deploy(charm)
+
+    await ops_test.model.wait_for_idle(status="active",  timeout=600)
+
+
+#@pytest.mark.skip_if_deployed
+#@pytest.mark.abort_on_fail
+async def test_action_update_testflinger(ops_test: OpsTest):
+    #charm = await ops_test.build_charm(CHARM_PATH)
+    #app = await ops_test.model.deploy(charm)
```

Comment:
same here with some declarations too

Reply:
Also fixed, thanks. It turns out the environment only needs to be deployed once. Which is nice, because it can take a long time in the integration tests.